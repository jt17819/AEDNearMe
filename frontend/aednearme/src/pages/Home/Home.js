import React, { Fragment } from 'react';
import { Grid, Container, Button, makeStyles } from '@material-ui/core';
import { useNavigate } from 'react-router';
import MainMap from '../../components/MainMap/MainMap';
import './Home.css';

const useStyles = makeStyles(() => ({

  title: {
    borderBottom: '1px solid #fff',
    paddingBottom: '5px',
  },

  text: {
    fontSize: '16px'
  },

  btn_box:{
    marginTop: "30px",
    display: 'flex',
    justifyContent: 'space-between',
  },

  primary: {
    border: "none",
    background: "#273746",
    color: '#fff',
    borderRadius: "0px",
    padding: "10px 20px",
    "&:hover": {
      background: "#34495E",
    }
  },
  
  secondary: {
    border: "none",
    background: "aliceblue",
    color: "#ba181b",
    borderRadius: "0px",
    padding: "10px 20px",
    "&:hover": {
      background: "aliceblue",
    }
  },

}));

const Home = () => {
  const navigate = useNavigate();

  const goToAEDhowTo = () => {
    navigate('/AEDhowTo')
  }

  const goToUpload = () => {
    navigate('/upload')
  }

  const classes = useStyles();


  

  return (
    <Fragment>

      <Container  
        style={{
          paddingBottom: "80px",
          paddingTop: "20px"
        }}>
        <Grid container spacing={4}>

          <Grid item md={5} >
            <h1 className={classes.title}
            >IN A LIFE THREATENING EMERGENCY, CALL 999 or 112</h1>
            
            <p className={classes.text}
            >Use our location driven database to show your nearest AED.
            </p>


            <p className={classes.text} >
            An AED machine, or Automated External Defibrillator, is an essential component in first aid and safety.
            </p>


            <p className={classes.text} >
            Most of us have seen these units at airports, hospitals, schools, government buildings, nursing homes, first aid stations and even in our workplaces â€” But do we remember where we seen them ?
            </p>

            <div className={classes.btn_box}>
              <Button 
                className={classes.primary}  
                onClick={goToAEDhowTo}
              >How to use AED</Button>

              <Button 
                className={classes.secondary} 
                onClick={goToUpload} 
              >Add new Location</Button>

            </div>
            
          </Grid>

          <Grid
            item
            md={7}
            xs={12}
            className=""
            >
            <div>
              <MainMap />
            </div>
          </Grid>

        </Grid>

      </Container>
          
    </Fragment>
  );
};

export default Home;
