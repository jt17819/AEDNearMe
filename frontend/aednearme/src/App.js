import React from "react";
import { Route, Routes, Outlet } from 'react-router-dom';
import Home from "./pages/Home/Home";
import AEDhowto from "./pages/AED/AEDhowto";
import CPRhowto from "./pages/CPR/CPRhowto";
import Quiz from "./pages/Quiz/Quiz";
import Choking from "./pages/Choking/Choking";

import Login from "./pages/Login/Login";
import Register from "./pages/Register/Register";
import Upload from "./pages/Upload/Upload";
import FAQ from "./pages/FAQ/FAQ";

import { NavBar } from "./components/NavBar/NavBar";
import { Footer } from "./components/Footer/Footer.js";

import './App.css';

function App() {
  return (
    <div className="App">
      <NavBar />
      <Routes>
          <Route path='/' element={<Home />} />
          <Route path='/aedhowto' element={<AEDhowto />} />
          <Route path='/cprhowto' element={<CPRhowto />} />
          <Route path='/quiz' element={<Quiz />} />
          <Route path='/choking' element={<Choking />} />
          <Route path='/login' element={<Login />} />
          <Route path='/register' element={<Register />} />
          <Route path='/upload' element={<Upload />} />
          <Route path='/faq' element={<FAQ />} />
   
          <Route path='*' element={<p>nothing to see here, move on mate.</p>} />
      </Routes>

      <Footer />

      <Outlet />
    </div>
  );
}

export default App;
