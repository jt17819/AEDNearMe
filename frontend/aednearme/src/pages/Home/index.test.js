import { default as Home } from './Home';
import App from '../../App'
import { screen, render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {MemoryRouter, Router, BrowserRouter} from 'react-router-dom'
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'


const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));


describe('Home Page', () => {
    let setPlayersMock;
    let handleChangeMock;

    // beforeEach(() => {
    //     // user = jest.fn();
    //     setPlayersMock = jest.fn();
    //     // render(<MemoryRouter><Home /></MemoryRouter>);
    // });


    test('it renders a heading', () => {
        render(<MemoryRouter><Home /></MemoryRouter>);
        let heading = screen.getByRole('heading');
        expect(heading).toBeInTheDocument();;
    });


      // Rendering

  it('renders welcome message', () => {
    render(<Home />, { wrapper: MemoryRouter })
    expect(screen.getByText('IN A LIFE THREATENING EMERGENCY, CALL 999 or 112')).toBeInTheDocument();
  });








  // Snapshot

  it('should take a snapshot', () => {
    // render(<Home />, { wrapper: MemoryRouter })
    const { asFragment } = render(<Home />, { wrapper: MemoryRouter })
    
    expect(asFragment(<Home />)).toMatchSnapshot()
  })




});