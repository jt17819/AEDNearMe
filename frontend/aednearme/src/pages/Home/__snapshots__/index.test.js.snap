// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root pb-5 MuiContainer-maxWidthLg"
    style="padding-bottom: 100px; padding-top: 20px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-5"
      >
        <h1
          class="makeStyles-title-11"
        >
          IN A LIFE THREATENING EMERGENCY, CALL 999 or 112
        </h1>
        <p
          class="makeStyles-text-12"
        >
          Use our location driven database to show your nearest AED.
        </p>
        <p
          class="makeStyles-text-12"
        >
          An AED machine, or Automated External Defibrillator, is an essential component in first aid and safety.
        </p>
        <p
          class="makeStyles-text-12"
        >
          Most of us have seen these units at airports, hospitals, schools, government buildings, nursing homes, first aid stations and even in our workplaces â€” But do we remember where we seen them ?
        </p>
        <div
          class="makeStyles-btn_box-13"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-primary-14"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              How to use AED
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-secondary-15"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add new Location
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <div>
          <div />
          <div>
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
